### PR TITLE
Generate PHP source code for tests

### PR DESCRIPTION
**Pull request by volter9**
Hey Kyle. I think it would be much easier to test the result if tester would also generate compiled PHP files (like cached ones). This PR adds into process of testing compiled PHP templates.

What do you think?

### CODE REVIEW

**Comment by kylekatarnls** [comment]
Hi, I'm currently working on unit tests on this branch: https://github.com/kylekatarnls/jade-php/tree/travis
I'm totally agree, those files should not be in the repository.

But this will not be compatible with the future merge of the travis branch. If I drop the branch, I will merge this PR, else, we'll see after the merge, I think we can do the test with no files to be written at all.

**Comment by volter9** [comment]
> I think we can do the test with no files to be written at all.

Of course, however it will require a quite different type of testing, like testing Lexer and Compiler separately. One buddy realized this like [that](https://github.com/dopesaur/dogescript-php/blob/master/tests/LexerTest.php).

File: .gitignore [hunk 1]
```diff
@@ -4,4 +4,7 @@
 
 # Composer
 /vendor
-/composer.lock
\ No newline at end of file
+/composer.lock
+
+# Tests
+/tests/*.compiled.php
```
Comment (kylekatarnls) [inline .gitignore]: A .gitignore is present in /tests/. If we do this, we will use this separated .gitignore

Comment (volter9) [inline .gitignore]: I didn't noticed .gitignore in `/tests` in the first place :)

**Comment by kylekatarnls** [comment]
No longer needed, thanks.

**Comment by volter9** [comment]
Cool :)

**Comment by kylekatarnls** [comment]
Hi, sorry, I just understand the original purpose of this PR. The problems I noticed:
- With services like Travis, we can not write files during tests AFAIK
- I do not understand your is_file test? Why not replace the cached file each time?

Instead of writing files, I suggest to print in the console the PHP code (with a flag option for example).

**Pull request closed without merging**